### PR TITLE
How to troubleshoot a build-time regression

### PR DESCRIPTION
**Pull request by jcouv**

### CODE REVIEW

File: docs/wiki/Troubleshooting-tips.md [hunk 1]
```diff
@@ -64,3 +64,19 @@ I recently had to test a [Roslyn change](https://github.com/dotnet/roslyn/pull/2
...
+There are three significant candidates to investigate:
+1. analyzer issue:  
+  Use `/p:ReportAnalyzer=true` to add analyzer timing information to the binary log.  
+  The binary log viewer can display that information.
+2. difference in inputs:  
+  Use `/p:Features=debug-determinism` to create an additional output file that documents all the inputs to a particular compilation.  
+  The file is written next to the compilation output and has a `.key` suffix.  
+  Comparing those files between slow and fast runs helps detect pertinent changes (new inputs, new references, etc).  
+3. compiler server issue:  
+  Inspect the binary log (search for "Error:" and such).
```
Comment (sharwell) [inline docs/wiki/Troubleshooting-tips.md:79]: 💡 There is a predefined (example) search term in MSBuild Structured Log Viewer for `$message CompilerServer failed` that I normally use for this.

Comment (jcouv) [reply docs/wiki/Troubleshooting-tips.md]: Good to know. Thanks!

**Comment by jcouv** [comment]
@jaredpar @dotnet/roslyn-compiler for review. Thanks

**Review by 333fred — APPROVED** [review]

File: docs/wiki/Troubleshooting-tips.md [hunk 2]
```diff
@@ -64,3 +64,19 @@ I recently had to test a [Roslyn change](https://github.com/dotnet/roslyn/pull/2
...
+
+There are three significant candidates to investigate:
+1. analyzer issue:  
+  Use `/p:ReportAnalyzer=true` to add analyzer timing information to the binary log.  
+  The binary log viewer can display that information.
+2. difference in inputs:  
+  Use `/p:Features=debug-determinism` to create an additional output file that documents all the inputs to a particular compilation.  
+  The file is written next to the compilation output and has a `.key` suffix.  
+  Comparing those files between slow and fast runs helps detect pertinent changes (new inputs, new references, etc).  
+3. compiler server issue:
```
Comment (jaredpar) [inline docs/wiki/Troubleshooting-tips.md:78]: This would be #2 for me.

**Pull request merged**